src/data/events/EventsD2ApiRepository.ts

```diff
@@ -158,8 +161,16 @@ export class EventsD2ApiRepository implements EventsRepository {
 
         return _(result)
             .flatten()
-            .map(object => ({ ...object, id: object.event }))
-            .map(object => cleanObjectDefault(object, defaults))
+            .map(object => {
+                const isUpdatedAfterStartDate = new Date(object.lastUpdated).toISOString() >= startDate.format();
```

Comment:
Typically, we'd directly compare Date or Moment objects. But no need to change it, comparing string date lexicographically will work too.